### PR TITLE
Pin pyzmq install version, other dependencies

### PR DESCRIPTION
**Pull request by anpaz**
To workaround: https://github.com/zeromq/pyzmq/issues/1764

### CODE REVIEW

**Review by cgranade — APPROVED** [review]

**Comment by anpaz** [comment]
Picks #719 and #718

**Comment by ricardo-espinoza** [comment]
Thanks @anpaz ... Is the change in 'nbconvert' and 'notebook' intended to address the alert on the 'mistune' package?

File: images/iqsharp-base/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,9 @@ RUN pip install --no-cache --upgrade pip
 RUN pip install \
         ipython==8.4.0 \
         jupyter==1.0.0 \
-        notebook==6.4.12
+        notebook==6.4.12 \
+        # to workaround: https://github.com/zeromq/pyzmq/issues/1764
+        pyzmq==23.2.1
```
Comment (ricardo-espinoza) [inline images/iqsharp-base/Dockerfile:12]: The build is failing on this installation.

Comment (anpaz) [reply images/iqsharp-base/Dockerfile:12]: this is the Dockerfile to build the docker image. Build is failing on another step.

**Review by cgranade — APPROVED** [review]

**Comment by anpaz** [comment]
@ricardo-espinoza, we're not updating `notebook`, just `nbconvert`, and yes, it is to address dependabot and `mistune` alert.

**Review by ricardo-espinoza — APPROVED** [review]

**Pull request merged**